clientv3/watch.go

```diff
@@ -371,6 +371,9 @@ func (w *watcher) Close() (err error) {
 			err = werr
 		}
 	}
+	if err == context.Canceled {
+		return fmt.Errorf("Context canceled as expected")
+	}
 	return err
```

Comment:
Are there situations where context would be canceled here but not expected?

Reply:
I think that's the case right now that it's canceled but not expected by end user. https://github.com/etcd-io/etcd/blob/master/clientv3/watch.go#L414

Reply:
I understand I just mean are there possible corner cases in your experience where this error message would not be appropriate. I am only asking because I have not dug into this.

Reply:
Thanks @hexfusion  So far playing with it, I believe this message covers both cases - one where it's displayed without user being explicitly canceling it (i.e. by calling Close which seems most common case) and second if there is anyway context was canceled by user (I wonder why anyone would do this but just thinking as a possible case).